source/server/listener_impl.cc

```diff
@@ -26,14 +26,81 @@
 namespace Envoy {
 namespace Server {
 
+ListenSocketFactoryImplBase::ListenSocketFactoryImplBase(
+    ListenerComponentFactory& factory, Network::Address::InstanceConstSharedPtr local_address,
+    Network::Address::SocketType socket_type, const Network::Socket::OptionsSharedPtr& options,
```

Comment:
This socket factory is a member of ListenerConfig, I would like not to reference to its parent because the time when this socket factory is created, ListenerConfig itself might not be fully populated. And only listener_name  is from ListenerConfig, the rest parameters are temporary variable in the context.

Reply:
yes, ListenerConfig doesn't have all params needed right now.

>/**
> * ListenSocketFactory is used to create listening socket for each listener.
> * Listeners created from the same ListenConfig instance share the same
> * ListenSocketFactory instance.
> */

According to `ListenSocketFactory`'s doc string, looks like it makes more sense to use ListenerConfig to construct a ListenSocketFactory. The missed parameters can be provided by ListenerConfig by adding new APIs to it.